cordova-lib/src/plugman/fetch.js

```diff
@@ -70,6 +70,10 @@ function fetchPlugin(plugin_src, plugins_dir, options) {
             //if --fetch was used, throw error for subdirectories
             if (result[2] && options.fetch) {
                 return Q.reject(new CordovaError('--fetch does not support subdirectories'));
+            } else {
+                if(result[2]) {
+                    events.emit('warn', 'support for subdirectories is depreated and will be removed in Cordova@7');
+                }
```

Comment:
Without knowing anything, I was just going by what the PR intended to do: "warns of deprecation of subdirectories". Therefore, if that flag is there, it should print out the warning. If the intent was different (in case of other conditionals), that should be made clear.